text/0000-trits-t5b1/0000-trits-t5b1.md

```diff
@@ -0,0 +1,119 @@
+    [ -1, 1, -1, 0, -1 ],  [ 0, 1, -1, 0, -1 ],   [ 1, 1, -1, 0, -1 ],    [ -1, -1, 0, 0, -1 ],  [ 0, -1, 0, 0, -1 ],
+    [ 1, -1, 0, 0, -1 ],   [ -1, 0, 0, 0, -1 ],   [ 0, 0, 0, 0, -1 ],     [ 1, 0, 0, 0, -1 ],    [ -1, 1, 0, 0, -1 ],
+    [ 0, 1, 0, 0, -1 ],    [ 1, 1, 0, 0, -1 ],    [ -1, -1, 1, 0, -1 ],   [ 0, -1, 1, 0, -1 ],   [ 1, -1, 1, 0, -1 ],
+    [ -1, 0, 1, 0, -1 ],   [ 0, 0, 1, 0, -1 ],    [ 1, 0, 1, 0, -1 ],     [ -1, 1, 1, 0, -1 ],   [ 0, 1, 1, 0, -1 ],
+    [ 1, 1, 1, 0, -1 ],    [ -1, -1, -1, 1, -1 ], [ 0, -1, -1, 1, -1 ],   [ 1, -1, -1, 1, -1 ],  [ -1, 0, -1, 1, -1 ],
+    [ 0, 0, -1, 1, -1 ],   [ 1, 0, -1, 1, -1 ],   [ -1, 1, -1, 1, -1 ],   [ 0, 1, -1, 1, -1 ],   [ 1, 1, -1, 1, -1 ],
+    [ -1, -1, 0, 1, -1 ],  [ 0, -1, 0, 1, -1 ],   [ 1, -1, 0, 1, -1 ],    [ -1, 0, 0, 1, -1 ],   [ 0, 0, 0, 1, -1 ],
+    [ 1, 0, 0, 1, -1 ],    [ -1, 1, 0, 1, -1 ],   [ 0, 1, 0, 1, -1 ],     [ 1, 1, 0, 1, -1 ],    [ -1, -1, 1, 1, -1 ],
+    [ 0, -1, 1, 1, -1 ],   [ 1, -1, 1, 1, -1 ],   [ -1, 0, 1, 1, -1 ],    [ 0, 0, 1, 1, -1 ],    [ 1, 0, 1, 1, -1 ],
+    [ -1, 1, 1, 1, -1 ],   [ 0, 1, 1, 1, -1 ],    [ 1, 1, 1, 1, -1 ],     [ -1, -1, -1, -1, 0 ], [ 0, -1, -1, -1, 0 ],
+    [ 1, -1, -1, -1, 0 ],  [ -1, 0, -1, -1, 0 ],  [ 0, 0, -1, -1, 0 ],    [ 1, 0, -1, -1, 0 ],   [ -1, 1, -1, -1, 0 ],
+    [ 0, 1, -1, -1, 0 ],   [ 1, 1, -1, -1, 0 ],   [ -1, -1, 0, -1, 0 ],   [ 0, -1, 0, -1, 0 ],   [ 1, -1, 0, -1, 0 ],
+    [ -1, 0, 0, -1, 0 ],   [ 0, 0, 0, -1, 0 ],    [ 1, 0, 0, -1, 0 ],     [ -1, 1, 0, -1, 0 ],   [ 0, 1, 0, -1, 0 ],
+    [ 1, 1, 0, -1, 0 ],    [ -1, -1, 1, -1, 0 ],  [ 0, -1, 1, -1, 0 ],    [ 1, -1, 1, -1, 0 ],   [ -1, 0, 1, -1, 0 ],
+    [ 0, 0, 1, -1, 0 ],    [ 1, 0, 1, -1, 0 ],    [ -1, 1, 1, -1, 0 ],    [ 0, 1, 1, -1, 0 ],    [ 1, 1, 1, -1, 0 ],
+    [ -1, -1, -1, 0, 0 ],  [ 0, -1, -1, 0, 0 ],   [ 1, -1, -1, 0, 0 ],    [ -1, 0, -1, 0, 0 ],   [ 0, 0, -1, 0, 0 ],
+    [ 1, 0, -1, 0, 0 ],    [ -1, 1, -1, 0, 0 ],   [ 0, 1, -1, 0, 0 ],     [ 1, 1, -1, 0, 0 ],    [ -1, -1, 0, 0, 0 ],
+    [ 0, -1, 0, 0, 0 ],    [ 1, -1, 0, 0, 0 ],    [ -1, 0, 0, 0, 0 ]
+]
+```
+
+Because the memory footprint (`1.2kb`) of this `LUT` can be too high for some very limited systems, we also present algorithms that do not use it.
+
+## Encoding
+
+<!-- TODO -->
+
+## Decoding
+
+### With LUT
+
+Decoding a `byte` into `5` `trits` with the help of the `LUT` is fairly simple:
+- if the `byte` is positive, use it as an index for the `LUT` e.g. if the byte is `42` `LUT[42] = [ 0, -1, -1, -1, 1 ]` and `0 * 3^0 + (-1) * 3^1 + (-1) * 3^2 + (-1) * 3^3 + 1 * 3^4 = 42`.
+
+### Without LUT
+
+<!-- TODO -->
+
+## Usual sizes
+
+For convenience purposes, here are some usual sizes and their encoded equivalents:
+
+- `27` `trits` fit in `6` `bytes`
+- `81` `trits` fit in `17` `bytes`
+- `243` `trits` fit in `49` `bytes`
+- `6561` `trits` fit in `1313` `bytes`
+- `8019` `trits` fit in `1604` `bytes`
+
+# Drawbacks
+
+<!-- TODO -->
+
+# Rationale and alternatives
+
+<!-- TODO -->
```

Comment:
What other operations should be supported for an encoding? If we consider `[u8]` as array of trit quintuples we might want to copy some subrange of trits that is not aligned to a bound of 5 trits (for example, copy trits from 2 to 13 encoded as bytes): 
```
fn copy(from: &[u8], from_offset: usize, to: &mut [u8], to_offset: usize, count: usize)
```
A trivial implementation would first decode `from` into trits and then encode subrange of the trits into `to` bytes. But it uses additional memory and two conversions (from and to bytes). A more efficient implementation would use no additional memory and one conversion via a more intricate LUT.